### ISSUE TITLE
BTC->ERC20, demo ready

### ISSUE THREAD

**Issue by D4nte** [comit-network/comit-rs#438]
Do BTC->ERC20 end-to-end test.

Note:
Write test step by step and fix as you go.
If anything major, fork to a new issue.

Child of #399.
Child of #408.
Blocked by #380.
Blocked by #435 
Blocked by #445
